### PR TITLE
fsverity-tests.py: Add fsverity test cases

### PR DESCRIPTION
**Pull request by nasastry**
fsverity tests verifies the integrity/authenticity associtated
with the file.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : 200c158bd27c7d2b4924fe99137f38581bf845e1
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-06T02.22-200c158/job.log
 (1/1) fsverity-tests.py:fsverity.test: PASS (1.94 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-06T02.22-200c158/results.html
JOB TIME   : 2.51 s

SLES:
JOB ID     : 23cdfce2c80dd34d549995323e0b80882f7bc34f
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-06T11.52-23cdfce/job.log
 (1/1) fsverity-tests.py:fsverity.test: PASS (1.76 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-06T11.52-23cdfce/results.html
JOB TIME   : 2.03 s


JOB ID     : 4015ecbb86ff0348f7f6cf325de8f2cc5efa3635
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-06T02.41-4015ecb/job.log
 (1/1) fsverity-tests.py:fsverity.test: CANCEL: CONFIG_FS_VERITY not set. (0.59 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-06T02.41-4015ecb/results.html
JOB TIME   : 1.13 s
```

File: security/fsverity-tests.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        deps = ['gcc', 'make']
+        if detected_distro.name in ['rhel', 'SuSE', 'fedora', 'centos']:
+            deps.extend(["autoconf", "automake", "openssl-devel"])
+        else:
+            self.cancel("Unsupported distro %s for fsverity package"
+                        % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        url = "https://git.kernel.org/pub/scm/linux/kernel/git/ebiggers/fsverity-utils.git"
```
Comment (abdhaleegit) [inline security/fsverity-tests.py:52]: better to also have url from yaml

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
apart from url rest all LGTM

**Comment by nasastry** [comment]
This test case run applies only to `upstream`, there is no `distro` package. 

```
RHEL:
JOB ID     : cd173e8c457a6efc3267a818a03fc276daf1acbd
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-19T03.26-cd173e8/job.log
 (1/1) fsverity-tests.py:fsverity.test: CANCEL: CONFIG_FS_VERITY not set. (0.45 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-19T03.26-cd173e8/results.html
JOB TIME   : 12.56 s


SLES:
JOB ID     : b4ba9de4810128de1c3f4ef21df608cbbe8add0b
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-19T13.54-b4ba9de/job.log
 (1/1) fsverity-tests.py:fsverity.test: PASS (3.80 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-19T13.54-b4ba9de/results.html
JOB TIME   : 25.09 s

```

[rhel_fsverity_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9818399/rhel_fsverity_job.log)
[sles_fsverity_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9818402/sles_fsverity_job.log)

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Pull request merged**